### PR TITLE
Deduplicate Youtube videoIds.

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: app/lib/service/youtube/backend.dart [hunk 1]
```diff
@@ -110,24 +113,33 @@ class _PkgOfWeekVideoFetcher {
...
                 (i.snippet?.description ?? '').trim().split('\n').first,
             thumbnailUrl: i.snippet!.thumbnails!.high!.url!,
-          ),
-        ));
+          );
+          if (videoIds.add(video.videoId)) {
+            videos.add(video);
+          } else {
+            _logger.warning(
+                'Duplicate videoId "${video.videoId}" on page $pageIndex.');
```
Comment (sigurdm) [inline app/lib/service/youtube/backend.dart:136]: Should we also log the previous page this occurred on?

**Review by sigurdm — COMMENTED** [review]
test?

File: app/lib/service/youtube/backend.dart (hunk 1 above)
Comment (isoos) [reply app/lib/service/youtube/backend.dart:136]: Do you think it is useful? We can't really use the information, not even the `pageIndex` much:
- the sort order is time descending, a new item will push the page indexes
- the pagination is based on continuation token.

**Comment by isoos** [comment]
> test?

We have a sanity-check local test with local dev key provided:
https://github.com/dart-lang/pub-dev/blob/master/app/test/service/youtube/backend_test.dart#L15-L20

**Comment by isoos** [comment]
Hm, actually, the test's expectation is not really great, because it has only a lower limit on the number of items. It seems we get much more than expected.

File: app/lib/service/youtube/backend.dart (hunk 1 above)
Comment (sigurdm) [reply app/lib/service/youtube/backend.dart:136]: Hmm - do we then want to log pageIndex at all?

Only thing I was curious about was if the duplicates could happen on the same page or not....

Comment (isoos) [reply app/lib/service/youtube/backend.dart:136]: Ah, that's a good point!

**Comment by isoos** [comment]
#5020 has the correct fix

**Pull request closed without merging**